### DISCUSSION TITLE
Deleting Multiple Objects via REST API

### DISCUSSION THREAD

**Discussion by CrazySliva** [netbox-community/netbox#7477] [Q&A] [answered]
Good afternoon.  Is it possible to delete multiple objects not through a method that requires a unique ID?
For example, I want to delete all virtual machines belonging to cluster 4  :

curl -s -X DELETE -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" https://yourdomain.com/api/virtualization/virtual-machines/?cluster=4

**Comment by jeremystretch** [comment]
[Bulk deletion via the REST API](https://netbox.readthedocs.io/en/stable/rest-api/overview/#deleting-multiple-objects) requires a unique ID to protect against race conditions, wherein the attributes of some objects are modified immediately prior to the deletion request. This approach ensures that _only_ the explicitly identified objects will be deleted.

**Comment by CrazySliva** [comment]
What are the workarounds for solving this case?
I've tried the following method:

1. Unloaded all virtual machines belonging to cluster 4 to a file.
curl -X GET "https://yourdomain.com/api/virtualization/virtual-machines/?cluster=4" -H "Accept: application / json; indent=4" -H "Authorization: Token $TOKEN" > temp.json

2. I tried to delete virtual machines 
curl -s -X DELETE -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" https://yourdomain.com/api/virtualization/virtual-machines/ --data @temp.json

But nothing happened, although the temp.json file contains the "id" of the field.

**Comment by drmsoffall** [comment, marked as answer]
If you really want to do this, you probably need to transform the JSON data with a tool like `jq`.  For example:
```
jq "[.results[] | {id}]" temp.json > temp_id.json
```
Then use the transformed data as the input to your bulk delete, rather than the original output.

**Reply by CrazySliva** [reply]
> If you really want to do this, you probably need to transform the JSON data with a tool like `jq`. For example:
> 
> ```
> jq "[.results[] | {id}]" temp.json > temp_id.json
> ```
> 
> Then use the transformed data as the input to your bulk delete, rather than the original output.

Thanks a lot! I wish you many rays of goodness.
